dbms/src/Storages/Page/V3/WALStore.cpp

```diff
@@ -201,9 +201,15 @@ bool WALStore::saveSnapshot(FilesSnapshot && files_snap, PageEntriesEdit && dire
     // Rename it to be a normal log file.
     const auto temp_fullname = log_filename.fullname(LogFileStage::Temporary);
     const auto normal_fullname = log_filename.fullname(LogFileStage::Normal);
+
     LOG_FMT_INFO(logger, "Renaming log file to be normal [fullname={}]", temp_fullname);
+    // Need link encryption info before rename
+    provider->linkEncryptionInfo(temp_fullname, normal_fullname);
+
     auto f = Poco::File{temp_fullname};
     f.renameTo(normal_fullname);
+    // Also need delete old encryption info after rename
+    provider->deleteEncryptionInfo(temp_fullname);
```

Comment:
We should use FileProvider::renameFile to replace line 206~212

https://github.com/pingcap/tiflash/blob/2b2750b7dfd7823be7c33e290492e1f8cffe1a76/dbms/src/Encryption/FileProvider.cpp#L183-L188

Reply:
I am sure about `renameFile ` from `FileProvider`. Because no used on `Page`.

But looks like better than `link` + `delete`